src/loader/bgw_launcher.c

```diff
@@ -306,8 +298,16 @@ start_db_schedulers(HTAB *db_htab)
 {
 	HASH_SEQ_STATUS hash_seq;
 	DbHashEntry *current_entry;
+	int			nstarted = 0;
+	int			ndatabases = hash_get_num_entries(db_htab);
 
-	/* now scan our hash table of dbs and register a worker for each */
+	/*
+	 * Increment our workers all at once so that schedulers don't start
+	 * workers stealing other schedulers' spots
+	 */
+	bgw_total_workers_increment_by(ndatabases);
```

Comment:
Shouldn't we check the return value here and return an error if the reservation didn't succeed?

Also, how do we deal with databases that are created once postgres is already running, which is true for pretty much all databases at some point?